src/include/sof/audio/module_adapter/intel/core/kernel/logger/log.h

```diff
@@ -0,0 +1,414 @@
+#endif
+
+EXTERN_C_BEGIN
+
+/*!
+ * \addtogroup skernel Scalable Kernel
+ * @{
+ *  \addtogroup logger Logger
+ *  @{
+ *    \addtogroup log_api API
+ *    @{
+ * This section describes generic logging API. Using LOG macro (and as
+ * derivatives also FLOG, ALOG, SLOG etc) in code will:
+ * 1. [COMPILATION] Create log_entry in .static_log section.
+ * 2. [COMPILATION] Check message format (should be printf-like).
+ * 3. [RUNTIME] Check whether message can be logged.
+ */
+
+#define LOGGER_THREADPTR_OFFSET 120
+
+#ifdef MODULE_NAME
+#define LOGGER_LIB_ID_VAR_NAME CONCAT(CONCAT(__, MODULE_NAME), _lib_id)
+extern const volatile uint32_t LOGGER_LIB_ID_VAR_NAME;
+#define LOGGER_LIB_ID_WRAP(input) (((uint32_t)input) | (LOGGER_LIB_ID_VAR_NAME << 3))
+#else
+#define LOGGER_LIB_ID_WRAP(input) input
+#endif
+
+/*! The following function pointers are used to put logs in different ways */
+/*! depending on selected logging backend */
+typedef void (*log_put_0_fn)(void*, uint32_t);
+typedef void (*log_put_1_fn)(void*, uint32_t, uint32_t);
+typedef void (*log_put_2_fn)(void*, uint32_t, uint32_t, uint32_t);
+typedef void (*log_put_3_fn)(void*, uint32_t, uint32_t, uint32_t, uint32_t);
+typedef void (*log_put_4_fn)(void*, uint32_t, uint32_t, uint32_t, uint32_t, uint32_t);
+typedef void (*log_put_5_fn)(void*, uint32_t, uint32_t, uint32_t, uint32_t, uint32_t, uint32_t);
+typedef void (*log_put_6_fn)(void*, uint32_t, uint32_t, uint32_t, uint32_t, uint32_t, uint32_t,
+				uint32_t);
+typedef void (*log_put_7_fn)(void*, uint32_t, uint32_t, uint32_t, uint32_t, uint32_t, uint32_t,
+				uint32_t, uint32_t);
+typedef void (*log_timer_callback_fn)(void *param);
+
+/*! Logger descriptor. */
+typedef struct logger {
+	struct log_transport *transport;
+	uint32_t log_en_mask_;
+	/*! pointers to functions to put logs */
+	/*! single function for different param count */
+	log_put_0_fn log_put_0;
+	log_put_1_fn log_put_1;
+	log_put_2_fn log_put_2;
+	log_put_3_fn log_put_3;
+	log_put_4_fn log_put_4;
+	log_put_5_fn log_put_5;
+	log_put_6_fn log_put_6;
+	log_put_7_fn log_put_7;
+
+	/*! amount of dropped messages */
+	uint32_t dropped_counter_;
+	/*! reference host time */
+	uint64_t host_time_reference_;
+	/*! reference dsp time*/
+	uint64_t dsp_time_reference_;
+
+	/*! Pointer to log buffer descriptor */
+	struct log_buffer *log_buffer;
+	/*! ACTUAL log buffer data capacity not log buffer size! */
+	uint32_t buffer_capacity;
+
+	uint32_t cached_last_ro_f;
+	uint32_t cached_wo;
+} logger_s;
+
+/*! Log level priority enumeration. */
+typedef enum log_priority {
+	/*! Critical message. */
+	L_CRITICAL = 0,
+	/*! Error message. */
+	L_ERROR = 0,
+	/*! High importance log level. */
+	L_HIGH = 1,
+	/*! Warning message. */
+	L_WARNING = 1,
+	/*! Medium importance log level. */
+	L_MEDIUM = 2,
+	/*! Low importance log level. */
+	L_LOW = 3,
+	/*! Information. */
+	L_INFO = 3,
+	/*! Verbose message. */
+	L_VERBOSE = 4,
+	L_DEBUG   = 4,
+	L_MAX     = 5,
+} AdspLogPriority,
```

Comment:
This is a problem of library build in IADK environment. It was build in IADK defined log priorities and library generates such priorities. I need to convert those into Zephyr definitions in system service call.  

Library calls this function:
void SystemServiceLogMessage(AdspLogPriority log_priority, uint32_t log_entry,
		AdspLogHandle const *log_handle, uint32_t param1, uint32_t param2,
		uint32_t param3, uint32_t param4);

Passing AdspLogPriority like in this define. In the SystemServiceLogMessage() routine body I can "translate" this into Zephyr codes and generate Zephyr kind log.  Can not change this directly since library does not know Zephyr log priority definitions.

Reply:
ok, I wonder if we can get away with converting some log levels using the compiler e.g.
```c
#define SOME_IADK_LOG_LEVEL EQUIVALENT_ZEPHYR_LOG_LEVEL
```
My expectation is that all legacy modules will need recompiled for use with Zephyr RTOS

Also thinking we could have a high level macro to exclude some definitions that are used by IADK legacy code e.g.
```c
#ifdef IADK_LEGACY_SUPPORT
#include "iadk_rtos_definitions.h"
#else 
#include "zephyr_rtos_definitions.h"
#endif
```
The IADK build CC cmd line could define ```-DIADK_LEGACY_SUPPORT``` and pull in the correct header of the older modules that need a lot of the older macros etc. The newer stuff could use the Zephyr definitions.

Reply:
Mapped to log levels already in SOF.  There is separate development to add cAVS/ACE legacy logging format support. Loadable modules logging will be available after this work will be done.